src/inputs/RichTextField.tsx

```diff
@@ -63,65 +57,59 @@ export function RichTextField(props: RichTextFieldProps) {
-        } else {
-          currentHtml.current = innerHTML;
-          const mentions = extractIdsFromMentions(mergeTags || [], textContent || "");
-          onChange && onChange(innerHTML, textContent || undefined, mentions);
+        currentHtml.current = value;
+        editor.current!.loadHTML(value || "");
+        // Remove listener once we've initialized
+        window.removeEventListener("trix-initialize", onEditorInit);
+
+        function trixChange(e: ChangeEvent) {
+          const { textContent, innerHTML } = e.target;
+          const onChange = onChangeRef.current;
+          // If the user only types whitespace, treat that as undefined
+          if ((textContent || "").trim() === "") {
+            currentHtml.current = undefined;
+            onChange && onChange(undefined, undefined, []);
+          } else {
+            currentHtml.current = innerHTML;
+            const mentions = extractIdsFromMentions(mergeTags || [], textContent || "");
+            onChange && onChange(innerHTML, textContent || undefined, mentions);
+          }
         }
+
+        const trixBlur = () => maybeCall(onBlurRef.current);
+        const trixFocus = () => maybeCall(onFocusRef.current);
+
+        // We don't want to allow file attachment for now.  In addition to hiding the button, also disable drag-and-drop
+        // https://github.com/basecamp/trix#storing-attached-files
+        const preventDefault = (e: any) => e.preventDefault();
+        window.addEventListener("trix-file-accept", preventDefault);
+
+        editorElement.current.addEventListener("trix-change", trixChange as any);
+        editorElement.current.addEventListener("trix-blur", trixBlur);
+        editorElement.current.addEventListener("trix-focus", trixFocus);
       }
+    }
 
-      // We don't want to allow file attachment for now.  In addition to hiding the button, also disable drag-and-drop
-      // https://github.com/basecamp/trix#storing-attached-files
-      const preventDefault = (e: any) => e.preventDefault();
-      window.addEventListener("trix-file-accept", preventDefault);
-
-      const trixBlur = () => maybeCall(onBlurRef.current);
-      const trixFocus = () => maybeCall(onFocusRef.current);
-
-      editorElement.addEventListener("trix-change", trixChange as any, false);
-      editorElement.addEventListener("trix-blur", trixBlur as any, false);
-      editorElement.addEventListener("trix-focus", trixFocus as any, false);
-
-      return () => {
-        editorElement.removeEventListener("trix-change", trixChange as any);
-        editorElement.removeEventListener("trix-blur", trixBlur as any);
-        editorElement.removeEventListener("trix-focus", trixFocus as any);
-        window.removeEventListener("trix-file-accept", preventDefault);
-      };
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [readOnly],
-  );
+    // Attaching listener to the `window` to we're listening prior to render.
```

Comment:
Maybe throw in a comment that this is specifically because web component events might fire before our `useEffect`s run, particularly when we're rendered in a portal?